### PR TITLE
Add FxCop with only ConfigureAwait rule enabled for now

### LINKED ISSUE #917: Consider adding FxCop
As suggested by @danielmarbach in [this comment](https://github.com/rabbitmq/rabbitmq-dotnet-client/pull/897#discussion_r453729376):

> Context capturing is applied to the continuation which is the end of the method. Though in general libs that don't need context capture should always opt out for every await. It would be best to even enable i.ex. Fxcop even if only https://docs.microsoft.com/en-us/visualstudio/code-quality/ca2007?view=vs-2019 is activated

### PR DESCRIPTION
**Pull request by danielmarbach**
## Proposed Changes

Fixes #917

## Types of Changes

What types of changes does your code introduce to this project?
_Put an `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes issue #NNNN)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause an observable behavior change in existing systems)
- [ ] Documentation improvements (corrections, new content, etc)
- [x] Cosmetic change (whitespace, formatting, etc)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating
the PR. If you're unsure about any of them, don't hesitate to ask on the
mailing list. We're here to help! This is simply a reminder of what we are
going to look for before merging your code._

- [x] I have read the `CONTRIBUTING.md` document
- [x] I have signed the CA (see https://cla.pivotal.io/sign/rabbitmq)
- [x] All tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in related repositories

### CODE REVIEW

**Review by michaelklishin — APPROVED** [review]

**Review by stebet — APPROVED** [review]
Like it!

File: projects/Unit/Unit.csproj [hunk 1]
```diff
@@ -4,6 +4,7 @@
     <TargetFrameworks>net461;netcoreapp3.1</TargetFrameworks>
     <AssemblyOriginatorKeyFile>../rabbit.snk</AssemblyOriginatorKeyFile>
     <SignAssembly>true</SignAssembly>
+    <RunAnalyzersDuringBuild>false</RunAnalyzersDuringBuild>
```
Comment (adamralph) [inline projects/Unit/Unit.csproj:7]: @danielmarbach why is this required? I see that no analyzers are installed in this project, nor in the parent `Directory.Build.targets`. Or is there a transitive dependency somewhere?

**Comment by danielmarbach** [comment]
I have no idea anymore. Without this I got warning in test project automatically because Fxcop seemed to be loaded fully and analysing all assemblies. At least this is how it was locally at that time in Rider. I can try to remove it again when I have time at some point if anybody cares

File: projects/Unit/Unit.csproj (hunk 1 above)
Comment (danielmarbach) [reply projects/Unit/Unit.csproj:7]: Without the flag in VS

image

Comment (danielmarbach) [reply projects/Unit/Unit.csproj:7]: It is related to `PrivateAssets="None"`. If switched to `All` we can remove the exclusion from analysis

Comment (danielmarbach) [reply projects/Unit/Unit.csproj:7]: Raised https://github.com/rabbitmq/rabbitmq-dotnet-client/pull/941

Comment (adamralph) [reply projects/Unit/Unit.csproj:7]: > `PrivateAssets="None"`

👍 yes, that explains it.
